test/qa/lib/firewall.py

```diff
@@ -280,20 +361,20 @@ def is_active(ports: list[Port] | None = None, subnets: list[str] | None = None)
     expected_rules = _get_firewall_rules(ports, subnets)
     print("\nExpected rules:")
     logging.log("\nExpected rules:")
-    for rule in expected_rules:
-        print(rule)
+    for key, rule in expected_rules.items():
+        print(f"{key}: \n{rule}")
         logging.log(rule)
 
-    current_rules = _get_iptables_rules()
+    current_rules = _get_iptables_rules(_get_all_interfaces())
     print("\nCurrent rules:")
     logging.log("\nCurrent rules:")
-    for rule in current_rules:
-        print(rule)
+    for key, rule in current_rules.items():
+        print(f"{key}: \n{rule}")
         logging.log(rule)
 
     print()
     print(sh.nordvpn.settings())
-    return current_rules == expected_rules
+    return all(current_rules[key] == expected_rules[key] for key in expected_rules)
```

Comment:
looks like we lose some order with this approach, would this rule order fail or pass the test? (this is not the intended rule order)
```
-A POSTROUTING -o wlp0s20f3 -m connmark --mark 0xe1f1 -m comment --comment nordvpn -j ACCEPT
-A POSTROUTING -o wlp0s20f3 -m comment --comment nordvpn -j DROP
-A POSTROUTING -o enp0s31f6 -m connmark --mark 0xe1f1 -m comment --comment nordvpn -j ACCEPT
-A POSTROUTING -o enp0s31f6 -m comment --comment nordvpn -j DROP